test/net_http_test.rb

```diff
@@ -312,6 +312,43 @@ def test_adding_custom_errors_do_trip_circuit
     end
   end
 
+  def test_5xxs_trip_circuit_when_fatal_server_flag_enabled
+    options = proc do |host, port|
+      {
+        tickets: 2,
+        success_threshold: 1,
+        error_threshold: 3,
+        error_timeout: 10,
+        open_circuit_server_errors: true,
+        name: "#{host}_#{port}",
+      }
+    end
+
+    with_semian_configuration(options) do
+      with_server do
+        http = Net::HTTP.new(HOSTNAME, PORT)
+        http.raw_semian_options[:error_threshold].times do
+          http.get("/500")
+        end
+        assert_raises Net::CircuitOpenError do
```

Comment:
As discussed earlier, what might be counter-intuitive (to some) about this is that normally on `5xx` you just receive a status code (not an exception). With this patch, you'll get exceptions on `5xx`. Most code (that handles connection failures) should handle this well—but some might not.

I don't think that's a huge deal, given you have to handle those conditions—but it's worth keeping in mind. I don't think people will find one ultimately more intuitive than the other, just want to make clear that this is a choice.